Source/Extensions/UITextFieldExtension.swift

```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+extension UITextField {
+
+    static func disableHardwareLayout() {
+#if targetEnvironment(simulator)
+        // Disable hardware keyboards.
+        let setHardwareLayout = NSSelectorFromString("setHardwareLayout:")
```

Comment:
This is for simulation keyboard. when user use numeric keypad using connect hardware keyboard then user can type non numeric numbers. It was a request from Deepesh on a meeting.

Reply:
I think I get it, you're forcing the appearance of the OS keyboard in simulator, so that keyboard input from machine running simulator cannot bypass inputType? If so will just resolve to cut off the noise